var/spack/repos/builtin/packages/argobots/package.py

```diff
@@ -46,4 +56,21 @@ def configure_args(self):
         else:
             args.append('--disable-debug')
 
+        if '+stackunwind' in self.spec:
+            args.append('--enable-stack-unwind')
+            args.append('--with-libunwind={0}'.format(self.spec['libunwind'].prefix))
+
+        if 'stackguard=canary' in self.spec:
```

Comment:
Curious.  Why not use `canary-32` as a value?  If you do, you could replace this `if`statement with something along the lines:

```suggestion
        stackguard = self.spec.variants['stackguard'].value
        if stackguard != 'no':
            args.append('--enable-stack-overflow-check={0}'.format(stackguard))
```

Reply:
@tldahlgren Thank you for your comment! I was afraid of using a value that contains `-`. Your suggested one is much cleaner.

I updated the `stackguard` variant as you suggested. Please check the second commit I added.